actioncable/lib/action_cable/connection/web_socket.rb

```diff
@@ -36,6 +36,27 @@ def rack_response
 
       private
         attr_reader :websocket
+
+      @driver_selector = nil
+      def self.create_driver(env, event_target, event_loop, protocols)
+        case @driver_selector
+        when :rack
+          puts "Remembered..."
+          return WebSocketRack.attempt(env, event_target, event_loop, protocols)
+        when :driver
+          return ::WebSocket::Driver.websocket?(env) && ClientSocket.new(env, event_target, event_loop, protocols)
+        end
+        return nil unless ::WebSocket::Driver.websocket?(env)
+        puts "Calculating memory..."
+        ret = WebSocketRack.attempt(env, event_target, event_loop, protocols)
+        if (ret)
+          @driver_selector = :rack
```

Comment:
Couldn't we just store a class here (`WebSocketRack` or `ClientSocket`) and convert `create_driver` into:

```ruby
def driver_class(env)
  @driver_class ||= begin

  end
end
```

And then use it in `initialize`: `@websocket = driver_class(env)&.new(env, event_target, ...)`.

And also replace `attempt(env, ...)` with `available?(env)`.

@boazsegev WDYT?